src/Lean/Util/Trace.lean

```diff
@@ -83,7 +83,7 @@ variable {α : Type} {m : Type → Type} [Monad m] [MonadTrace m] [MonadOptions
 
 def printTraces : m Unit := do
   for {msg, ..} in (← getTraceState).traces do
-    IO.println (← msg.format)
+    IO.println (← msg.format.toIO)
```

Comment:
Huh, I've never seen this cast being necessary before

Reply:
@Kha Yeah, this is a rare case caused by the `[MonadLiftT IO m]` instance parameter of this function.